libr/core/cmd_write.c

```diff
@@ -850,997 +844,1151 @@ static void cmd_write_pcache(RCore *core, const char *input) {
+
+			p = strtok (NULL, " ");
+			b_size = p && *p ? r_num_math (core->num, p) : 0;
+			if (dist != 0){
+				r_core_shift_block (core, addr, b_size, dist);
+				r_core_seek (core, addr, true);
+				cmd_suc = true;
+			}
 		}
+		free (input_shadow);
 		break;
-	case 'A': // "wA"
-		switch (input[1]) {
-		case ' ':
-			if (input[2] && input[3]==' ') {
-				r_asm_set_pc (core->rasm, core->offset);
-				eprintf ("modify (%c)=%s\n", input[2], input+4);
-				len = r_asm_modify (core->rasm, core->block, input[2],
-					r_num_math (core->num, input+4));
-				eprintf ("len=%d\n", len);
-				if (len>0) {
-					if (!r_core_write_at (core, core->offset, core->block, len)) {
+	case 'X': // "weX"
+		if (input[1] == ' ') {
+			addr = r_num_math (core->num, input+2);
+			input += 2;
+			while (*input && *input != ' ') input++;
+			input++;
+			len = *input ? strlen (input) : 0;
+			bytes = len > 1? malloc (len+1) : NULL;
+			len = bytes ? r_hex_str2bin (input, bytes) : 0;
+			if (len > 0) {
+				//ut64 cur_off = core->offset;
+				cmd_suc = r_core_extend_at (core, addr, len);
+				if (cmd_suc) {
+					if (!r_core_write_at (core, addr, bytes, len)) {
 						cmd_write_fail (core);
 					}
-					WSEEK (core, len);
-				} else eprintf ("r_asm_modify = %d\n", len);
-			} else eprintf ("Usage: wA [type] [value]\n");
-			break;
-		case '?':
-		default:
-			r_core_cmd_help (core, help_msg_wA);
-			break;
-		}
-		break;
-	case 'c': // "wc"
-		switch (input[1]) {
-		case '\0': // "wc"
-			//if (!r_config_get_i (core->config, "io.cache"))
-			//	eprintf ("[warning] e io.cache must be true\n");
-			r_io_cache_list (core->io, 0);
-			break;
-		case '?': // "wc?"
-			r_core_cmd_help (core, help_msg_wc);
-			break;
-		case '*': // "wc*"
-			r_io_cache_list (core->io, 1);
-			break;
-		case '+': // "wc+"
-			if (input[2]=='*') { // "wc+*"
-				//r_io_cache_reset (core->io, core->io->cached);
-				eprintf ("TODO\n");
-			} else if (input[2]==' ') { // "wc+ "
-				char *p = strchr (input+3, ' ');
-				ut64 to, from;
-				from = r_num_math (core->num, input+3);
-				if (p) {
-					*p = 0;
-					to = r_num_math (core->num, input+3);
-					if (to<from) {
-						eprintf ("Invalid range (from>to)\n");
-						return 0;
-					}
 				} else {
-					to = from + core->blocksize;
+					eprintf ("r_io_extend failed\n");
 				}
-				r_io_cache_commit (core->io, from, to);
-			} else {
-				eprintf ("Invalidate write cache at 0x%08"PFMT64x"\n", core->offset);
-				r_io_cache_commit (core->io, core->offset, core->offset+1);
+				core->offset = addr;
+				r_core_block_read (core);
 			}
-			break;
-		case '-': { // "wc-"
-			if (input[2]=='*') { // "wc-*"
-				r_io_cache_reset (core->io, core->io->cached);
-				break;
+			free (bytes);
+		}
+		break;
+	case '?': // "we?"
+	default:
+		cmd_suc = false;
+		break;
+	}
+	if (cmd_suc == false) {
+		r_core_cmd_help (core, help_msg_we);
+	}
+	return 0;
+}
+
+static int wp_handler_old(void *data, const char *input) {
```

Comment:
i would prefix all those static functions as cmd_wp() or `cmd_wp_old` (maybe 'old' is not the best worst either, maybe `string` `str` or `cmd_wp_strarg` ..

Reply:
`old` is fine since eventually will be removed anyway.

Reply:
I think `_old` is also inline with `OLDINPUT` which is used in r_cmd.h to indicate the "old"(aka current) handlers which parse input themselves.